src/domain/flow-system/services/src/flow/active_configs_state.rs

```diff
@@ -101,6 +101,46 @@ impl ActiveConfigsState {
             .get(BorrowedFlowKeyDataset::new(dataset_id, flow_type).as_trait())
             .copied()
     }
+
+    pub fn get_merged_config_snapshot(
```

Comment:
There is no merging happening here, just overrides.

Reply:
The method is a bit unclear. Maybe we should return the configuration snapshot from the stored settings, rather than combining it with unrelated argument snapshot? Combining should better live in the service.

Reply:
Changed